plugin-mainmenu/xdgcachedmenu.cpp

```diff
@@ -73,6 +73,7 @@ XdgCachedMenu::XdgCachedMenu(MenuCache* menuCache, QWidget* parent): QMenu(paren
 {
     // qDebug() << "CREATE MENU FROM CACHE" << menuCache;
     MenuCacheDir* dir = menu_cache_get_root_dir(menuCache);
+    menu_cache_desktop_env_flag_ = menu_cache_get_desktop_env_flag(menuCache, "LXQt:X-LXQt:LXQT:X-LXQT");
```

Comment:
>Shouldn't the session name be taken from some *XDG_*\* env variable?

XDG_CURRENT_DESKTOP it is. Although string above may be valid too, but
only as fallback, in case if $XDG_CURRENT_DESKTOP is not set.